k8s-install/canal.yaml

```diff
@@ -0,0 +1,218 @@
+                  key: etcd_endpoints
+            # Disable Calico BGP.  Calico is simply enforcing policy. 
+            - name: CALICO_NETWORKING
+              value: "false"
+            # Disable file logging so `kubectl logs` works.
+            - name: CALICO_DISABLE_FILE_LOGGING
+              value: "true"
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: /lib/modules
+              name: lib-modules
+              readOnly: true
+            - mountPath: /var/run/calico
+              name: var-run-calico
+              readOnly: false
+        # This container installs the Calico CNI binaries
+        # and CNI network config file on each node.
+        - name: install-calico-cni
+          image: calico/cni:v1.4.1
+          command: ["/install-cni.sh"]
+          env:
+            # The location of the etcd cluster.
+            - name: ETCD_ENDPOINTS
+              valueFrom:
+                configMapKeyRef:
+                  name: canal-config
+                  key: etcd_endpoints
+            # The CNI network config to install on each node.
+            - name: CNI_NETWORK_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: canal-config
+                  key: cni_network_config
+          volumeMounts:
+            - mountPath: /host/opt/cni/bin
+              name: cni-bin-dir
+            - mountPath: /host/etc/cni/net.d
+              name: cni-net-dir
+      volumes:
+        # Used by calico/node.
+        - name: lib-modules
+          hostPath:
+            path: /lib/modules
+        - name: var-run-calico
+          hostPath:
+            path: /var/run/calico
+        # Used to install CNI.
+        - name: cni-bin-dir
+          hostPath:
+            path: /opt/cni/bin
+        - name: cni-net-dir
+          hostPath:
+            path: /etc/cni/net.d
+        # Used by flannel daemon.
+        - name: run-flannel
+          hostPath:
+            path: /run/flannel
+        - name: etcd-certs 
+          hostPath:
+            path: /etc/ssl/etcd
+        - name: certs
+          hostPath:
+            path: /usr/share/ca-certificates
+        - name: resolv
+          hostPath:
+            path: /etc/resolv.conf
+
+---
+
+# This manifests deploys a Job which performs one time 
+# configuration of Canal. 
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: configure-canal 
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    k8s-app: canal 
+spec:
+  template:
+    metadata:
+      name: configure-canal
+    spec:
+      hostNetwork: true
+      restartPolicy: OnFailure 
+      containers:
+        # Writes basic flannel configuration to etcd. 
+        - name: configure-flannel
+          image: quay.io/coreos/etcd:v3.0.9 
+          command: 
+          - "etcdctl"
+          - "set"
+          - "/coreos.com/network/config"
+          - '{ "Network": "192.168.0.0/16" }'
```

Comment:
I'd like to expose this as a configuration parameter, but it's lower priority for now.